honeybadgermpc/protocols/avid.py

```diff
@@ -0,0 +1,216 @@
+                response_set.add(sender)
+
+                if len(response_set) >= response_threshold:
+                    try:
+                        m = decode_output(roothash)
+                    except Exception as e:
+                        logger.info(f"Failed to decode message: {e}")
+                        m = ""
+                    return m
+
+    async def disperse(self, sid, pid, input_list):
+        """ Main information dispersal handling
+        :param int sid: e.x. tag to be used
+        :param int pid: current member id
+        :param int input_list: the list of message
+            for each party from 1 to n from the dealer
+        """
+        # setup some parameters
+        k = self.n - 2 * self.f  # Wait to reconstruct.
+        echo_threshold = math.ceil((self.n + self.f + 1) / 2)  # number of ECHOs needed
+        ready_threshold = self.f + 1  # number of READYs needed for READY
+        output_threshold = 2 * self.f + 1  # number of READYs needed for storing
+
+        def broadcast(o):
+            for i in range(self.n):
+                self.send(i, o)
+
+        # leader handling
+        if pid == self.leader:
+            # The leader erasure encodes the input, sending one strip to each participant
+            assert len(input_list) == self.input_size
+
+            # construct the stripe list which contains the stripes for party 1 to party n
+            stripes_list = [None] * self.input_size
+            mt_list = [None] * self.input_size
+            roothash_list = [None] * self.input_size
+            for i, m in enumerate(input_list):
+                stripes_list[i] = encode(k, self.n, m)
+                mt_list[i] = merkle_tree(stripes_list[i])
+                roothash_list[i] = mt_list[i][1]
+
+            # transpose for sending
+            stripes_list_per_party = [list(i) for i in zip(*stripes_list)]
+            for i in range(self.n):
+                branch_list = [None] * self.input_size
+                for j in range(self.input_size):
+                    branch_list[j] = get_merkle_branch(i, mt_list[j])
+                # send each person the column of stripes
+                self.send(i, (sid, 'VAL', roothash_list,
+                              branch_list, stripes_list_per_party[i]))
+
+        # counters and variables for msg
+        echo_set = set()
+        ready_set = set()
+        ready_sent = False
+        from_leader = None
+        # internal storage
+        my_stripes = [None] * self.input_size
+        my_roothash_list = [None] * self.input_size
+
+        while True:  # main recv loop
+            sender, msg = await self.recv()
+            if msg[1] == 'VAL' and from_leader is None:
+                # Validation
+                (_, _, roothash_list, branch_list, stripes_for_each) = msg
+                if sender != self.leader:
+                    logger.warning(
+                        f"[{pid}] VAL message from other than leader: {sender}")
```

Comment:
same here, use `%s` and not f"{pid}", please fix in all places in the code.

Reply:
Resolved.